pkg/metrics/install.go

```diff
@@ -0,0 +1,134 @@
+package metrics
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"regexp"
+	"time"
+
+	"github.com/pkg/errors"
+	kotsv1beta1 "github.com/replicatedhq/kots/kotskinds/apis/kots/v1beta1"
+	"github.com/replicatedhq/kots/pkg/buildversion"
+	"github.com/segmentio/ksuid"
+)
+
+const (
+	DevEndpoint  = "http://localhost:30016"
+	ProdEndpoint = "https://replicated.app"
+)
+
+type InstallMetrics struct {
+	endpoint    string
+	InstallID   string    `json:"install_id"`
+	StartedAt   time.Time `json:"started"`
+	FinishedAt  time.Time `json:"finished"`
+	FailedAt    time.Time `json:"failed"`
+	KotsVersion string    `json:"kots_version"`
+	Cause       string    `json:"cause"`
+}
+
+func InitInstallMetrics(license *kotsv1beta1.License, disableOutboundConnections bool) *InstallMetrics {
+	if disableOutboundConnections {
+		return nil
```

Comment:
`InstallMetrics` has exported fields, which can't be private because of json tags... So I feel it would be better to keep `disableOutboundConnections` as an internal flag instead of returning nil here.